### PR TITLE
[fix](sort) fix nullable column sorting incorrectly

### LINKED ISSUE #13124: [Bug] (daily) the nullable column is sorted incorrectly
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

a daily failed case: https://ci-builds.apache.org/job/Doris/job/doris_daily_enable_vectorized/70/testReport/junit/(root)/test_query_join/test_left_outer_join/

```
mysql> select a.k1, b.k1, a.k2, a.k3 from test a left join baseall b on a.k2 = b.k2 and a.k1 > 0   where a.k1 = -118 and a.k2 < 2500 and a.k2 > 1500   order by 1, 2, 3, 4;
+------+------+------+-------------+
| k1   | k1   | k2   | k3          |
+------+------+------+-------------+
| -118 | NULL | 1710 | -1723139760 |
| -118 | NULL | 2150 |  1854253526 |
| -118 | NULL | 2308 |  -756946701 |
| -118 | NULL | 1989 |  1973272318 |
+------+------+------+-------------+
```

### What You Expected?

```
mysql> select a.k1, b.k1, a.k2, a.k3 from test a left join baseall b on a.k2 = b.k2 and a.k1 > 0   where a.k1 = -118 and a.k2 < 2500 and a.k2 > 1500   order by 1, 2, 3, 4;
+------+------+------+-------------+
| k1   | k1   | k2   | k3          |
+------+------+------+-------------+
| -118 | NULL | 1710 | -1723139760 |
| -118 | NULL | 1989 |  1973272318 |
| -118 | NULL | 2150 |  1854253526 |
| -118 | NULL | 2308 |  -756946701 |
+------+------+------+-------------+
```

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by luozenglin**
# Proposed changes

Issue Number: close #13124

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by HappenLee — DISMISSED** [review]
LGTM

File: be/src/vec/core/sort_block.h [hunk 1]
```diff
@@ -259,15 +261,25 @@ class ColumnSorter {
...
                     }
                     if (is_null_range.first < is_null_range.second) {
+                        // do not sort null values
                         std::fill(flags.begin() + is_null_range.first,
-                                  flags.begin() + is_null_range.second, 1);
+                                  flags.begin() + is_null_range.second, 0);
 
-                        flags[is_null_range.first] = 0;
+                        if (_limit > is_null_range.first && _limit <= is_null_range.second) {
+                            _limit = is_null_range.second;
```
Comment (Gabriel39) [inline be/src/vec/core/sort_block.h:269]: `_limit > is_null_range.first && _limit <= is_null_range.second` will be true only once so I think this line is trival and can be removed

Comment (luozenglin) [reply be/src/vec/core/sort_block.h:269]: Why can this statement be removed? I think an `UNLIKELY` could be added.

**Review by xy720 — DISMISSED** [review]
LGTM

**Review by yangzhg — APPROVED** [review]
lgtm

**Pull request merged**